python/cudf/cudf/_lib/pylibcudf/groupby.pyx

```diff
@@ -0,0 +1,245 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.
+
+from cython.operator cimport dereference
+from libcpp.functional cimport reference_wrapper
+from libcpp.memory cimport unique_ptr
+from libcpp.pair cimport pair
+from libcpp.utility cimport move
+from libcpp.vector cimport vector
+
+from cudf._lib.cpp.groupby cimport (
+    aggregation_request,
+    aggregation_result,
+    groupby,
+    groups,
+    scan_request,
+)
+from cudf._lib.cpp.scalar.scalar cimport scalar
+from cudf._lib.cpp.table.table cimport table
+from cudf._lib.cpp.types cimport size_type
+
+from .aggregation cimport Aggregation
+from .column cimport Column
+from .table cimport Table
+from .types cimport null_policy, sorted
+from .utils cimport _as_vector
+
+
+cdef class GroupByRequest:
+    """A request for a groupby aggregation or scan.
+
+    Parameters
+    ----------
+    values : Column
+        The column to aggregate.
+    aggregations : list
+        The list of aggregations to perform.
+    """
+    def __init__(self, Column values, list aggregations):
+        self.values = values
+        self.aggregations = aggregations
+
+    cdef aggregation_request to_libcudf_agg_request(self) except *:
+        """Convert to a libcudf aggregation_request object.
+
+        This method is for internal use only. It creates a new libcudf
```

Comment:
In general should these be prefixed with `_`?

Reply:
Yeah I think so.